app/gui2/src/util/ast/abstract.ts

```diff
@@ -168,50 +170,48 @@ export function parseQualifiedName(ast: Ast): QualifiedName | null {
 /** Substitute `pattern` inside `expression` with `to`.
  * Will only replace the first item in the property acccess chain. */
 export function substituteIdentifier(
-  module: MutableModule,
-  expression: Ast,
+  expr: MutableAst,
   pattern: IdentifierOrOperatorIdentifier,
   to: IdentifierOrOperatorIdentifier,
 ) {
-  const expr = module.getVersion(expression) ?? expression
-  if (expr instanceof Ident && expr.code() === pattern) {
-    expr.updateValue(() => Ast.parse(to, module))
-  } else if (expr instanceof PropertyAccess && expr.lhs instanceof Ident) {
+  if (expr instanceof MutableIdent && expr.code() === pattern) {
+    expr.setToken(to)
+  } else if (expr instanceof MutablePropertyAccess && expr.lhs != null) {
```

Comment:
The `!= null` check should be inside the `instanceof PropertyAccess` conditional--a `PropertyAccess` can be an operator section like `.foo` and we still shouldn't change the RHS (needs another test).